### PR TITLE
feat: add DatabaseSize check

### LINKED ISSUE #2: [Check] Introduce check to detect large DB tables

### PR DESCRIPTION
**Pull request by alexloetscher95**
resolves https://github.com/w-vision/PimcoreMonitorBundle/issues/2

Default warning threshold: 512 MB
Default failure threshold: 1024 MB

Ok?

### CODE REVIEW

File: src/PimcoreMonitorBundle/Check/DatabaseSize.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+     */
+    private function getDatabaseSize(): int
+    {
+        $query = "SELECT SUM(data_length + index_length) AS size
+                    FROM information_schema.TABLES
+                    GROUP BY table_schema";
+        $size = $this->connection->fetchAll($query);
+
+        if(is_array($size) && isset($size[0]['size']))
+        {
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/Check/DatabaseSize.php]:
```suggestion
        if (\is_array($size) && isset($size[0]['size'])) {
```

File: src/PimcoreMonitorBundle/Check/DatabaseSize.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            return (int)$size[0]['size'];
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/Check/DatabaseSize.php]:
```suggestion
            return (int) $size[0]['size'];
```

File: src/PimcoreMonitorBundle/Check/DatabaseSize.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+        if ($size >= $this->criticalThreshold) {
+            return new Failure(\sprintf('Database size is too high: %s', \formatBytes($size)), $data);
+        }
+
+        if ($size >= $this->warningThreshold) {
+            return new Warning(\sprintf('Database size is high: %s', \formatBytes($size)), $data);
+        }
+
+        return new Success(sprintf('Database size is %s', \formatBytes($size)), $data);
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/Check/DatabaseSize.php]:
```suggestion
        return new Success(\sprintf('Database size is %s', \formatBytes($size)), $data);
```

**Review by aarongerig — CHANGES_REQUESTED** [review]
@alexloetscher95 Some minor code style changes, but other than that OK.
Are 512MB / 1024MB reasonable size thresholds?

**Comment by alexloetscher95** [comment]
> @alexloetscher95 Some minor code style changes, but other than that OK. Are 512MB / 1024MB reasonable size thresholds?

@aarongerig I think that we should continually test and adapt this

**Comment by aarongerig** [comment]
@alexloetscher95 Code looks fine now, thanks!
What do you think about setting the warning threshold to something around 920MB, which would be around 90% of 1024MB  (error threshold). Otherwise I'm afraid that we get a lot of DBs with warnings, even though we don't need to take action.

Other than that, I approve this PR!

**Review by aarongerig — APPROVED** [review]

**Comment by alexloetscher95** [comment]
@aarongerig yeah. 90% makes much more sense :/ 
Changed it to 920 MB (approx. 90%) and will merge it immediately

**Pull request merged**